src/main/java/org/zalando/putittorest/zmon/ZmonResponseInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package org.zalando.putittorest.zmon;
+
+import org.apache.http.HttpException;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpResponseInterceptor;
+import org.apache.http.protocol.HttpContext;
+import org.zalando.zmon.actuator.metrics.MetricsWrapper;
+
+import java.io.IOException;
+
+import static java.lang.System.currentTimeMillis;
+
+public class ZmonResponseInterceptor implements HttpResponseInterceptor {
+
+    private final MetricsWrapper metricsWrapper;
+
+    public ZmonResponseInterceptor(final MetricsWrapper metricsWrapper) {
+        this.metricsWrapper = metricsWrapper;
+    }
+
+    @Override
+    public void process(final HttpResponse response, final HttpContext context) throws HttpException, IOException {
+        final long endTime = currentTimeMillis();
+        final Timing timing = (Timing) context.getAttribute(Timing.ATTRIBUTE);
```

Comment:
Intention is that it should not fail hard if you only register the response interceptor, but probably it should because this is a configuration error.